src/building/bootstrapping.md

```diff
@@ -2,24 +2,39 @@
 
 <!-- toc -->
 
+[Bootstrapping][boot] is the process of using a compiler to compile a later version
+of itself.
 
-[*Bootstrapping*][boot] is the process of using a compiler to compile itself.
-More accurately, it means using an older compiler to compile a newer version
-of the same compiler.
+This raises a [chicken-or-the-egg
+paradox](https://en.wikipedia.org/wiki/Chicken_or_the_egg): what rust compiler
+was used to compile the very first rust compiler? The answer is that the first
+compiler was not written in rust. It was [written in OCaml][ocaml-compiler]. Of
+course, it has long been discarded and since then the only compiler that is
+able to compile some version of `rustc` is a slightly earlier version of
+`rustc`.
 
-This raises a chicken-and-egg paradox: where did the first compiler come from?
-It must have been written in a different language. In Rust's case it was
-[written in OCaml][ocaml-compiler]. However it was abandoned long ago and the
-only way to build a modern version of rustc is a slightly less modern
-version.
+For this purpose a python script `x.py` is provided at the root of the
+repository. `x.py` downloads a pre-compiled compiler—the stage 0 compiler—and
+with it compiles from the current source code a compiler—the stage 1 compiler.
+Additionaly, it may use the stage 1 compiler to compile from the current source
+code another compiler—the stage 2 compiler. Below describes this process in
+some detail, including the reason for a stage 2 compiler and more.
 
-This is exactly how `x.py` works: it downloads the current beta release of
-rustc, then uses it to compile the new compiler.
+## The stages of bootstrapping
 
-## Stages of bootstrapping
+Each stage involves:
+- An existing compiler and its set of dependencies.
+- Targets ([objects][objects]): `std` and `rustc`.
 
-Compiling `rustc` is done in stages. Here's a diagram, adapted from Joshua Nelson's
-[talk on bootstrapping][rustconf22-talk] at RustConf 2022, with detailed explanations below.
+Note: the compiler of a stage—e.g. "the stage 1 compiler"—refers to the
+compiler that is compiled at that stage, not the one that already exists.
+
+In the first stage (stage 0) the compiler is usually obtained by downloading a
```

Comment:
*usually* makes me wonder if other ways should be mentioned or linked to

Reply:
I agree. But I'm not familiar with other ways. And this doesn't seem like an issue created in this PR. Also, I've changed it to "typically". Resolve?